searchcore/src/tests/proton/persistenceengine/resource_usage_tracker/resource_usage_tracker_test.cpp

```diff
@@ -2,23 +2,40 @@
 
 #include <vespa/persistence/spi/resource_usage_listener.h>
 #include <vespa/persistence/spi/resource_usage.h>
+#include <vespa/searchcore/proton/attribute/attribute_usage_stats.h>
+#include <vespa/searchcore/proton/attribute/i_attribute_usage_listener.h>
 #include <vespa/searchcore/proton/persistenceengine/resource_usage_tracker.h>
 #include <vespa/searchcore/proton/test/disk_mem_usage_notifier.h>
 #include <vespa/vespalib/gtest/gtest.h>
 #include <vespa/vespalib/util/idestructorcallback.h>
+#include <atomic>
 
+using storage::spi::AttributeResourceUsage;
 using storage::spi::ResourceUsage;
 using proton::test::DiskMemUsageNotifier;
+using proton::AttributeUsageStats;
 using proton::DiskMemUsageState;
 using proton::ResourceUsageTracker;
 
 namespace {
 
 struct MyResourceUsageListener : public storage::spi::ResourceUsageListener
 {
-    using storage::spi::ResourceUsageListener::ResourceUsageListener;
+    std::atomic<size_t> _update_count;
 
+    MyResourceUsageListener()
+        : storage::spi::ResourceUsageListener(),
+          _update_count(0u)
+    {
+    }
+
+    void update_resource_usage(const ResourceUsage& resource_usage) override {
+        storage::spi::ResourceUsageListener::update_resource_usage(resource_usage);
+        ++_update_count;
+    }
     std::vector<double> get_usage_vector() const { return { get_usage().get_disk_usage(), get_usage().get_memory_usage() }; }
```

Comment:
Consider using ResourceUsage() instead, with the constructor that takes only disk and memory.